### ISSUE TITLE
Failed to decode some engine struct

### ISSUE THREAD

**Issue by na-ji** [trumank/uesave-rs#55]
Hello!

I'm currently using the lib as a crate on my project, to decode a GVAS save for the game Coral Island. Since a recent update on the save format, a new field that I can't decode was added. 
I also dumped the C++ headers using UEDumper, which was a great help to find the types of the others fields. The field I can't decode is `creatorUniqueId` of the struct `PlayerCharacterData`. 
```c++
/// Struct /Script/ProjectCoral.C_PlayerCharacterData
/// Size: 0x0160 (352 bytes) (0x000000 - 0x000160) align 8 MaxSize: 0x0160
struct FC_PlayerCharacterData
{ 
	class AC_PlayerCharacter*                          Character;                                                  // 0x0000   (0x0008)  
	FC_PlayerInformation                               playerInfo;                                                 // 0x0008   (0x0068)  
	FC_CharCuzStruct                                   playerAppearance;                                           // 0x0070   (0x00B8)  
	FUniqueNetIdRepl                                   creatorUniqueId;                                            // 0x0128   (0x0028)  
	class APlayerState*                                currentPlayer;                                              // 0x0150   (0x0008)  
	FC_PlayerCharacterId                               playerCharacterId;                                          // 0x0158   (0x0004)  
	unsigned char                                      UnknownData00_7[0x4];                                       // 0x015C   (0x0004)  MISSED
};
```

It has a type of `FUniqueNetIdRepl`, which seems to be a UUID used in the engine's online subsystem. 

```c++
/// Struct /Script/Engine.UniqueNetIdRepl
/// Size: 0x0028 (40 bytes) (0x000001 - 0x000028) align 8 MaxSize: 0x0028
struct FUniqueNetIdRepl : FUniqueNetIdWrapper
{ 
	unsigned char                                      UnknownData00_8[0x17];                                      // 0x0001   (0x0017)  MISSED
	TArray<char>                                       ReplicationBytes;                                           // 0x0018   (0x0010)  
};
```

So When I try to decode the save, it breaks while reading `creatorUniqueId` and overflowing to the next field, `playerCharacterId`. This is the error:
```
Error: at offset 17655021: unknown property type: "playerCharacterId"

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:116:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/backtrace.rs:331:13
   3: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
             at /Users/thenaji27/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.79/src/backtrace.rs:27:14
   4: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/result.rs:1989:27
   5: uesave::main
             at ./uesave_cli/src/main.rs:238:24
   6: core::ops::function::FnOnce::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
   7: std::sys::backtrace::__rust_begin_short_backtrace
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/sys/backtrace.rs:152:18
   8: std::rt::lang_start::{{closure}}
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/rt.rs:162:18
   9: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:284:13
  10: std::panicking::try::do_call
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:557:40
  11: std::panicking::try
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:521:19
  12: std::panic::catch_unwind
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panic.rs:350:14
  13: std::rt::lang_start_internal::{{closure}}
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/rt.rs:141:48
  14: std::panicking::try::do_call
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:557:40
  15: std::panicking::try
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:521:19
  16: std::panic::catch_unwind
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panic.rs:350:14
  17: std::rt::lang_start_internal
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/rt.rs:141:20
  18: std::rt::lang_start
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/rt.rs:161:17
  19: _main
```

Here is the save file:
[coral island.sav.zip](https://github.com/user-attachments/files/18049307/coral.island.sav.zip)

And here are the types I'm using right now (as a crate):
```rust
			types.add(String::from(".saveData.timeLocationMapTracker.Key"), StructType::DateTime);
            types.add(String::from(".saveData.timeLocationMapTracker.Value"), StructType::Vector);
            types.add(String::from(".saveData.interiorSaveDataMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.interiorSaveDataMap.tileEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.interiorSaveDataMap.tileEntryMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.interiorSaveDataMap.tileEntryMap.wallObjects.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.furnitureSaveDataMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.furnitureSaveDataMap.savedEntryMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.furnitureSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.NPCSaveData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.NPCSaveData.triggeredDynamicDialog.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.npcWorldSpeechCooldownMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.npcWorldSpeechCooldownMap.waypointTriggerMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.shopStocks.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.shopStocks.stockItems.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.shopStocks.limitedItemStock.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.dayEndEventCooldownMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.unlockedTornPages.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.mailRuntimeTags.Value"), StructType::GameplayTagContainer);
            types.add(String::from(".saveData.questRuntimeTags.Value"), StructType::GameplayTagContainer);
            types.add(String::from(".saveData.mineProgressionMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.labUpgradeData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.donatedItemInfo"), StructType::Struct(Some(String::from("Set"))));
            types.add(String::from(".saveData.dynamicQuestDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.enchantmentSaveData.enchantmentSaveDataMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.offeringGroupsMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.offeringGroupsMap.offeringsMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.divingOfferingMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.procToGrowSaves.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.divingLevelData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.divingLevelData.clusterData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.caughtFish.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.lastGiftFromNPCDate.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.minigameClaimedReward.claimedRewardWinPosition.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.itemStorageDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.gridObjectsDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmHouseMaterials.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.animalsData.logAnimalMoods.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.animalsData.logAnimalMoods.Value"), StructType::Struct(Some(String::from("Set"))));
            types.add(String::from(".saveData.ranchBuildingData.autoCollectorData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.interiorSaveData.tileEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.interiorSaveData.tileEntryMap.Value"), StructType::Struct(Some(String::from("Map"))));
            types.add(String::from(".saveData.ranchBuildingData.interiorSaveData.tileEntryMap.wallObjects.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.placedData.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.ranchBuildingData.placedData.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.gachaMachineData.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.tileSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.tileSaveDataMap.savedEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.tileSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmTileSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmTileSaveDataMap.savedEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.farmTileSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.playerStatistics.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.playerMasteryLevel.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.caughtBug.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.journalData.fishingCaughtables.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.journalData.catchingCaughtables.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.weekNpcHangoutData.Value"), StructType::Struct(Some(String::from("int32"))));
            types.add(String::from(".saveData.players.farmHouse.interiorSaveData.interiorTileData.tileEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.players.farmHouse.interiorSaveData.interiorTileData.tileEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.farmHouse.interiorSaveData.placedTileData.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.farmHouse.interiorSaveData.furnitureTileData.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.interiorTileData.tileEntryMap.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.interiorTileData.tileEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.interiorTileData.tileEntryMap.wallObjects.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.furnitureTileData.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.furnitureTileData.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.underwaterFarmHouseInterior.placedTileData.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.museumCollectionProgress.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.playerCharacterId"), StructType::Struct(None));
            types.add(String::from(".saveData.players.timeLocationMapTracker.Key"), StructType::Struct(None));
            types.add(String::from(".saveData.players.timeLocationMapTracker.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.shopStocks.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.dynamicQuestConditionDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.players.creatorUniqueId"), StructType::Struct(None));
            types.add(String::from(".saveData.dynamicQuestConditionDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.cropSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.cropSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.cropSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.debrisSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.debrisSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.debrisSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmBuildingSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmBuildingSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.farmBuildingSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitPlantSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitPlantSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitPlantSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitTreeSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitTreeSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fruitTreeSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.grassSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.grassSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.grassSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.grassSaveDataMap.savedEntryMap.entryArray.tuftInQuadrantMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.groundSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.groundSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.groundSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.lootSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.lootSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.lootSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.placedSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.placedSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.placedSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fenceSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fenceSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.gateSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.gateSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.floorSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.floorSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.treeSaveDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.treeSaveDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.treeSaveDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.disturbedSoilDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.disturbedSoilDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.disturbedSoilDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.giantCropDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.giantCropDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fertilizerDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fertilizerDataMap.savedEntryMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.fertilizerDataMap.savedEntryMap.entryArray.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.excavationSoilDataMap.Value"), StructType::Struct(None));
            types.add(String::from(".saveData.excavationSoilDataMap.savedEntryMap.Value"), StructType::Struct(None));
```

So if you have an idea on how to handle this field, that would be great 😄 

Thank you!

**Comment by trumank** [comment]
Should be fixed by 8ce64ce7272ce7da884a92ecbfd065de9d6a83fd. The ID seems to be empty (which makes sense given a net ID has no purpose being saved anyway) so I've only handled the case of size == 0. If you find another save with with a non-empty ID I can look at handling that case.

ref https://github.com/EpicGames/UnrealEngine/blob/1308e62273a620dd4584b830f6b32cd8200c2ad3/Engine/Source/Runtime/Engine/Private/OnlineReplStructs.cpp#L54

**Comment by na-ji** [comment]
Thank you for your quick help! That was so fast. Here is a save with a non-empty ID. 

[coral island.sav.zip](https://github.com/user-attachments/files/18053429/coral.island.sav.zip)

**Comment by trumank** [comment]
Thanks for the test case. Fully implemented in 2e60cb10c17a7c0f0ab70d525a6bb72f73b0dd32

**Comment by na-ji** [comment]
Thank you for everything @trumank, your project is awesome! It's so cool to have support for less common types like this one.
